ApiSecurityInAction/dotnet/NatterApi/Controllers/UserController.cs

```diff
@@ -1,11 +1,18 @@
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
 using Microsoft.AspNetCore.Mvc;
+using NatterApi.Extensions;
 using NatterApi.Models;
 using NatterApi.Models.Requests;
 using NatterApi.Services;
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
 
 namespace NatterApi.Controllers
 {
     [ApiController, Route("/users")]
+    [ValidateAntiForgeryToken]

```

Comment:
I think this is more a feature of MVC than it is of general CSRF attacks. This should be paired with an [`@Html.AntiForgeryToken`](https://docs.microsoft.com/en-us/aspnet/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages#generating-the-tokens) in a Razor page to generate a token of the kind he mentions in Section 4.4.2 _Has-based double-submit cookies_ in the first paragraph. In the second paragraph he goes on to say "an API does not have the luxury of adding hidden form fields to requests because most API clients want JSON or another data form rather than HTML."

Also, because this is an MVC feature, we don't actually get the `ValidateAntiforgeryToeknAuthorizationFilter` injected into the service collection (we don't use [one any of the `AddMvc` options](https://docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?view=aspnetcore-5.0#aspnet-core-antiforgery-configuration) and instead use `AddControllers` and `UseEndpoints`. This means that when you attempt to create a space, you get a 500 error instead of a 401.

I think [this section on Angular](https://docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?view=aspnetcore-5.0#angularjs) is more what we're looking for, since it checks for the header instead of the form data.